src/auto_contiguous.cpp

```diff
@@ -64,7 +64,10 @@ void auto_contiguous::apply(module& m) const
         if(ins->outputs().empty() and ins != last)
             continue;
         shape s = ins->get_shape();
-        if(not s.dynamic() and not s.standard() and s.elements() != 0)
+        // If s is not standard layout or has out of sequence strides, insert "contiguous" op
+        // to make a standard shape
+        if(not s.dynamic() and (not s.transposed() or s.normalize_standard() != s) and
```

Comment:
This wrong, it should be `not s.standard() or s.normalize_standard() != s`, that is replace `transposed` with `standard`.

Reply:
Blame it on a garbled `git merge`.  Fixed.

Reply:
Actually, thinking about it more, we could probably simplify this to just `s.as_standard() != s`, so the entire `if` statement becomes `if(not s.dynamic() and s.as_standard() != s and s.elements() != 0)`.

Reply:
Ran `make check` successfully on my machine but somehow in the commit and CI process here on Github, `test_ref` failed for both variations, both times in non-obvious ways.  Back to the drawing board for now.